src/loader.rs

```diff
@@ -258,6 +264,7 @@ impl<'a> LoaderInternal<'a> {
 
         Ok(LoaderInternal {
             ctx,
+            object,
```

Comment:
For the case of Mach-O, this may be the `object` corresponding to the `.dSYM`, whereas we probably want `--section` to be using sections from the `Mach-O`.

Reply:
I tried to fix this in commit dac56ae31d42c24a7f1f42ab884352b923549605 - can you have a look?